src/Command/Environment/EnvironmentDeleteCommand.php

```diff
@@ -197,51 +198,67 @@ function ($environment) {
                 ));
                 if ($questionHelper->confirm('Are you sure you want to delete this environment?')) {
                     $toDeactivate[$environmentId] = $environment;
-                    if (!$input->getOption('no-delete-branch')
-                        && $this->shouldWait($input)
-                        && ($input->getOption('delete-branch')
-                            || (
-                                $input->isInteractive()
-                                && $questionHelper->confirm("Delete the remote Git branch too?")
-                            )
-                        )) {
-                        $toDeleteBranch[$environmentId] = $environment;
-                    }
                 } else {
                     $error = true;
                 }
-            } elseif ($environment->status === 'inactive') {
-                if ($questionHelper->confirm("Are you sure you want to delete the remote Git branch <comment>$environmentId</comment>?")) {
+            }
+
+            // Ask about deleting the branch, which requires either an inactive
+            // environment, or waiting for it to be deactivated.
+            if (!$input->getOption('no-delete-branch')
+                && ($environment->status === 'inactive' || (isset($toDeactivate[$environmentId]) && $this->shouldWait($input)))) {
+                $message = isset($toDeactivate[$environmentId])
+                    ? "Delete the inactive environment (Git branch) too?"
+                    : "Are you sure you want to delete the inactive environment (Git branch) <comment>$environmentId</comment>?";
+                if ($input->getOption('delete-branch') || ($input->isInteractive() && $questionHelper->confirm($message))) {
                     $toDeleteBranch[$environmentId] = $environment;
-                } else {
+                } elseif (!isset($toDeactivate[$environmentId])) {
                     $error = true;
                 }
-            } elseif ($environment->status === 'dirty') {
-                $this->stdErr->writeln("The environment <error>$environmentId</error> is currently building, and therefore can't be deleted. Please wait.");
-                $error = true;
-            } elseif ($environment->status === 'deleting') {
-                $this->stdErr->writeln("The environment <info>$environmentId</info> is already being deleted.");
-            } else {
-                $this->stdErr->writeln("The environment <error>$environmentId</error> has an unrecognised status <error>" . $environment->status . "</error>.");
-                $error = true;
+                $needNewline = true;
             }
         }
 
+        if ($needNewline) {
+            $this->stdErr->writeln('');
+        }
+
         if (empty($toDeleteBranch) && empty($toDeactivate)) {
-            if ($error) {
-                $this->stdErr->writeln('');
-            }
             $this->stdErr->writeln('No environment(s) to delete.');
```

Comment:
A minor thing, but I'd suggest this for clarity while you're in there already ([reasoning](https://developers.google.com/style/plurals-parentheses?hl=en)):
```suggestion
            $this->stdErr->writeln('No environments to delete.');
```

Reply:
Agreed, thanks - although I want to make another PR with some wording changes for delete vs. deactivate as well